app/models/spree/order.rb

```diff
@@ -130,6 +130,14 @@ def states
       where("state != ?", state)
     }
 
+    # All the states an order can be in before completing the checkout
+    PRIOR_TO_COMPLETION_STATES = %w(cart address delivery payment).freeze
+    # All the states of a complete order but that shouldn't count towards the balance. Those that
+    # the customer won't enjoy.
+    NON_FULFILLED_STATES = %w(canceled returned).freeze
+
+    scope :in_incomplete_state, -> { where(state: PRIOR_TO_COMPLETION_STATES) }
```

Comment:
Could we go for a simpler name here? We could have `FULFILLED_STATES` and `UNFULFILLED_STATES` and the scope could be `unfulfilled`?

Also, you've only used this scope in one place where you explicitly use the negation of it with `not`, so maybe the opposite scope is more useful, eg: `fulfilled`?

Reply:
yeah, that looks better. Thanks!

Reply:
I think finalized states _should_ include complete though.

Reply:
The scope you're trying to get (below) is "all the states that are not cart, address, delivery, payment", which should be: complete, canceled, returned.

Reply:
"not `in_incomplete_state`" == "not `unfinalized`" == "`finalized`" (including complete)

Reply:
Ah, I see you want a scope that just includes only canceled or returned in some cases. Maybe that's a third scope?

I think semantically, if we have a scope called "finalized", it has to include the completed state.

Reply:
I'll find another name then